### ISSUE TITLE
Allow filtering on Woningnet "loting" listings

### ISSUE THREAD

**Issue by edo-codes** [wtfloris/hestia#55]
It would be useful if we could filter the results of Woningnet to only show those with
```py
res["PublicatieModel"] == "Lotingmodel"
```
for those of us not blessed with the necessary amount of waitlist years.

I'm not sure if the best way to go about this would be to add a new filter or to split the Woningnet agencies into lottery and non-lottery.

I'm willing to create a PR (though I'm not sure how to test it myself)

**Comment by OmriSteiner** [comment]
To be fair, I think adding more filters is a good idea - and it's okay that they're applicable for a subset of the agencies.
But this would require a small refactor of the parsing. I would add an interface which is aware of the various rental properties, i.e implementers could possibly add more properties such as size etc. These could then be filtered on.

**Comment by wtfloris** [comment]
I've been hesitant to add more global filters because this adds a parameter for all websites. You can see how this grows exponentially: 10 websites with 4 parameters (url, address, price and city) is only 40 parsed parameters in total. Add 2 parameters (e.g. amount of rooms and total area), and you now only need 3 extra websites to double the total amount of parsed parameters to 78.

As websites get modified and APIs change without notice, it's a matter of time before your total amount of parsed parameters becomes large enough for something to break on a weekly or even daily basis. If people can't count on Hestia to work well on their favourite websites then they have to keep checking them manually, at which point Hestia is useless. So it comes down to a reliability vs. feature set tradeoff, where reliability is much more important in my opinion.

Additionally, unless we can parse these new parameters for 100% of listings on all websites, people are going to be left wondering why some homes do have the total area and some don't. And how do you handle homes that don't have it? Do you send it to people that have filters set because otherwise they might miss something? Or do you not send it because you don't want to send something incorrect? And whichever one you choose, people will think the filtering is broken (and complain about it).

tldr: I think it adds too much complexity and maintenance for too little gain, but please convince me if you disagree

That being said (to get back on topic on this specific question): I think adding a filter for loterij/waitlist homes on WoningNet is a good addition. The WoningNet websites are all the same so in practice this only adds a single parsed parameter. I'll write something for this and test it in the dev env.

**Comment by ventilaar** [comment]
I agree that we should not add too many filters in the bot since that would make scraping more complicated while also being more of a maintenance burden. It's really not that hard to just open up the link and quickly glance at the property if it's suited for your needs or not.
